### PR TITLE
fix(helmfile): url joiner is broken, use upath instead

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Changes

change url-join to upath.

## Context

https://github.com/renovatebot/renovate/discussions/30295

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/helmfile/extract.ts [hunk 1]
```diff
@@ -118,12 +118,12 @@ export async function extractPackageFile(
 
       if (isOCIRegistry(dep.chart)) {
         res.datasource = DockerDatasource.id;
-        res.packageName = joinUrlParts(repoName, depName);
+        res.packageName = join(repoName, depName);
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:121]: I don't like that workaround, it will not work on windows

Comment (PhilipAbed) [reply lib/modules/manager/helmfile/extract.ts:121]: should i use `upath.posix.join`?

Comment (PhilipAbed) [reply lib/modules/manager/helmfile/extract.ts:121]: or is there something im not familiar with? please elaborate what should we do here :D

Comment (PhilipAbed) [reply lib/modules/manager/helmfile/extract.ts:121]: i will return it to + '/' + then its best

File: lib/modules/manager/helmfile/extract.ts [hunk 2]
```diff
@@ -118,12 +117,12 @@ export async function extractPackageFile(
 
       if (isOCIRegistry(dep.chart)) {
         res.datasource = DockerDatasource.id;
-        res.packageName = joinUrlParts(repoName, depName);
+        res.packageName = repoName + '/' + depName;
```
Comment (secustor) [inline lib/modules/manager/helmfile/extract.ts:120]: I'm pretty sure we do this all over the place for docker references 🤔 

Have you checked that?  
In that case this should be new utility and should not be part of manager logic

Comment (PhilipAbed) [reply lib/modules/manager/helmfile/extract.ts:120]: I'm just rolling back to how it was before i changed it and caused it to fail,
If you would like to open an issue to refactor later be my guest.
for now let's make our user happy and solve this.

Comment (viceice) [reply lib/modules/manager/helmfile/extract.ts:120]:
```suggestion
        res.packageName = `${repoName}/${depName}`;
```
use string template

File: lib/modules/manager/helmfile/extract.ts [hunk 3]
```diff
@@ -118,12 +117,12 @@ export async function extractPackageFile(
...
       } else if (registryData[repoName]?.oci) {
         res.datasource = DockerDatasource.id;
         const alias = registryData[repoName]?.url;
         if (alias) {
-          res.packageName = joinUrlParts(alias, depName);
+          res.packageName = alias + '/' + depName;
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:125]:
```suggestion
          res.packageName = `${alias}/${depName}`;
```

**Review by viceice — APPROVED** [review]

**Pull request merged**